### ISSUE TITLE
weekly build incorrect version number

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#1482]
**Describe the bug**
the most recent release (17th Jan) version should be `0.5.dev2103`
but it's `0.5` https://pypi.org/project/monai-weekly/#history

**Additional context**
-  config https://github.com/Project-MONAI/MONAI/blob/master/.github/workflows/weekly-preview.yml
- log https://github.com/Project-MONAI/MONAI/actions/runs/490995349

**Comment by wyli** [comment]
rerunning the pipeline pushed the correct version to pypi, I've removed the wrong releases. I couldn't locate the root cause but would be great to revise the config to print more detailed logs https://github.com/Project-MONAI/MONAI/blob/master/.github/workflows/weekly-preview.yml

**Comment by IsaacYangSLA** [comment]
@wyli , did you mean that the same weekly-preview.yml did push a correct version when you re-ran the pipeline?  I am trying to see if there is any possibility a race condition occurred.  Also, I believe that pipeline was not running in self-hosted runners.  I might need to run same pipeline in another repo to duplicate such issue.

**Comment by wyli** [comment]
yes rerunning the pipeline worked and the log of previous errors gone... @IsaacYangSLA any possibility that this line's `date` cmd returns None? https://github.com/Project-MONAI/MONAI/blob/edbad4305a942b8fe170134673a08044bc04a967/.github/workflows/weekly-preview.yml#L32

I'll paste more info here if I see the issue again

**Comment by IsaacYangSLA** [comment]
@wyli, as you suggested, I added some checks on the result of $(date +'%y%U') to avoid wrong format.  I will submit the PR soon.

**Comment by wyli** [comment]
thanks @IsaacYangSLA , incidentally I see the same versioning error from your branch https://github.com/IsaacYangSLA/MONAI/runs/1715608291?check_suite_focus=true#step:6:53

**Comment by IsaacYangSLA** [comment]
@wyli , thanks for the log.  I couldn't find out why the tag information was wrong.  I tried the updated yaml file which displayed more details in my repo (https://github.com/IsaacYangSLA/MONAI/actions/runs/504928407 and https://github.com/IsaacYangSLA/MONAI/actions/runs/504947813) and it seemed at least we should be able to get some debug log in case things go wrong.  I also mocked the YEAR_WEEK var to an empty string so the pipeline stopped (https://github.com/IsaacYangSLA/MONAI/actions/runs/504950783).  I think this would at least prevent publishing wrong version to PyPI.
